components/layout/flow.rs

```diff
@@ -1115,6 +1115,24 @@ impl BaseFlow {
         }
     }
 
+    /// Update the 'flags' field when computed styles have changed.
+    ///
+    /// These flags are initially set during flow construction.  They only need to be updated here
+    /// if they are based on properties that can change triggering `RECONSTRUCT_FLOW`.
+    pub fn update_flags(&mut self, style: &ServoComputedValues) {
+        // For absolutely-positioned flows, changes to top/bottom/left/right can cause these flags
+        // to get out of date:
+        if self.flags.contains(IS_ABSOLUTELY_POSITIONED) {
+            let logical_position = style.logical_position();
+            self.flags.set(INLINE_POSITION_IS_STATIC,
+                logical_position.inline_start == LengthOrPercentageOrAuto::Auto &&
+                logical_position.inline_end == LengthOrPercentageOrAuto::Auto);
+            self.flags.set(BLOCK_POSITION_IS_STATIC,
+                logical_position.block_start == LengthOrPercentageOrAuto::Auto &&
+                logical_position.block_end == LengthOrPercentageOrAuto::Auto);
+        }
```

Comment:
How about making this a new incremental reflow flag, along the lines of `RECONSTRUCT_FLOW` or `REFLOW_OUT_OF_FLOW`?